packages/mobile/src/account/Profile.tsx

```diff
@@ -17,29 +18,44 @@ import { Screens } from 'src/navigator/Screens'
 import { TopBarTextButton } from 'src/navigator/TopBarButton'
 import { StackParamList } from 'src/navigator/types'
 import PictureInput from 'src/onboarding/registration/PictureInput'
+import { saveImageDataUrlToFile } from 'src/utils/image'
 
 type Props = StackScreenProps<StackParamList, Screens.Profile>
 
 function Profile({ navigation, route }: Props) {
   const { t } = useTranslation(Namespaces.accountScreen10)
   const [newName, setNewName] = useState(useSelector(nameSelector) ?? '')
-  const [newPicture, setNewPicture] = useState(useSelector(pictureSelector) ?? '')
+  const picturePath = useSelector(pictureSelector)
+  const [newPictureUri, setNewPictureUri] = useState(picturePath)
 
   const dispatch = useDispatch()
 
   useEffect(() => {
     if (route.params?.save) {
       navigation.setParams({ save: false })
-      dispatch(setPicture(newPicture))
       dispatch(setName(newName))
-      // TODO: Save these things on CIP8.
+      dispatch(setPicture(newPictureUri))
+      // TODO: Save name and picture on CIP-8.
       dispatch(showMessage(t('namePictureSaved')))
       navigation.goBack()
+
+      // Delete old proflie pictures if necessary.
+      if (picturePath && picturePath !== newPictureUri) {
+        RNFS.unlink(picturePath).catch()
+      }
     }
   }, [route.params?.save])
 
-  const onPictureChosen = (chosenPicture: string) => {
-    setNewPicture(chosenPicture)
+  const onPictureChosen = (pictureDataUrl: string | null) => {
+    if (!pictureDataUrl) {
+      setNewPictureUri(null)
+    } else {
+      const newPicturePath = saveImageDataUrlToFile(
+        pictureDataUrl,
+        `file://${RNFS.DocumentDirectoryPath}/profile-${Date.now()}`
```

Comment:
nit: do you think it make sense to have a separate directory for profile photos?